### PR TITLE
chore: remove async

### PR DESCRIPTION
**Pull request by Vid201**
There is a `RwLock` for mempools and reputations on the UoPoolService level (`Arc<RwLock<HashMap<MempoolId...`), and both will always be accessed over these two hashmaps, thus we don't need async traits for Mempool and Reputation.

### CODE REVIEW

File: src/uopool/memory_mempool.rs [hunk 1]
```diff
@@ -14,75 +10,70 @@ use super::Mempool;
 #[derive(Default, Educe)]
 #[educe(Debug)]
 pub struct MemoryMempool {
-    user_operations: Arc<RwLock<HashMap<UserOperationHash, UserOperation>>>, // user_operation_hash -> user_operation
-    user_operations_by_sender: Arc<RwLock<HashMap<Address, HashSet<UserOperationHash>>>>, // sender -> user_operations
+    user_operations: HashMap<UserOperationHash, UserOperation>, // user_operation_hash -> user_operation
+    user_operations_by_sender: HashMap<Address, HashSet<UserOperationHash>>, // sender -> user_operations
```
Comment (zsluedem) [inline src/uopool/memory_mempool.rs:14]: I am curious what is the intention to remove the `RwLock` primitives here?

**Review by zsluedem — COMMENTED** [review]
Good work. I want to remove the async in these traits, too. Glad that we want to do the same. I left one question about the primitives`RwLock`

File: src/uopool/memory_mempool.rs (hunk 1 above)
Comment (Vid201) [reply src/uopool/memory_mempool.rs:14]: I don't think they are needed and bring some additional overhead. We have another `RwLock` here: https://github.com/Vid201/aa-bundler/blob/main/src/uopool/services/uopool.rs#L25. Mempools are always accessed through that `RwLock`, so the whole mempool should be anyway thread-safe.

What do you think?

Comment (zsluedem) [reply src/uopool/memory_mempool.rs:14]: Oh. That makes sense. If there are some performance problem in the future, we could make lock scope smaller at that time. Cool.

**Review by zsluedem — APPROVED** [review]
Great

File: src/uopool/memory_mempool.rs (hunk 1 above)
Comment (Vid201) [reply src/uopool/memory_mempool.rs:14]: Exactly, great!

**Pull request merged**